multiuser/keycloak/che-multiuser-keycloak-user-remover/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakTokenRequester.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2012-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.keycloak.server;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static javax.ws.rs.HttpMethod.POST;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import javax.inject.Singleton;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.commons.lang.IoUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Requests the Keycloak access token using a Keycloak username, password, and token endpoint. */
+@Singleton
+public class KeycloakTokenRequester {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KeycloakTokenRequester.class);
+
+  private static final String GRANT_TYPE = "grant_type";
+  private static final String GRANT_TYPE_VALUE = "password";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+  private static final String CLIENT_ID = "client_id";
+  private static final String CLIENT_ID_VALUE = "admin-cli";
+  private static final String ACCESS_TOKEN = "access_token";
+
+  /**
+   * Requests the Keycloak access token for the provided Keycloak user, password, and token
+   * endpoint.
+   *
+   * @param keycloakUser the Keycloak user
+   * @param keycloakPassword the Keycloak password
+   * @param keycloakTokenEndpoint the Keycloak token endpoint
+   * @return Keycloak access token
+   * @throws ServerException when any server error occurs while performing the request or while
+   *     parsing the response
+   */
+  public String requestToken(
+      String keycloakUser, String keycloakPassword, String keycloakTokenEndpoint)
+      throws ServerException {
```

Comment:
I think that is misbehaving usage of `org.eclipse.che.api.core.Exception` what if this method thow IOException | JsonSyntaxException ex?

Reply:
I'm not sure if I understand, do you suggest that I throw a different exception here?
https://github.com/eclipse-che/che-server/blob/9a6ffcbee9884d253c2c199e1e52b54a39380cde/multiuser/keycloak/che-multiuser-keycloak-user-remover/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakTokenRequester.java#L105-L108

Reply:
Originally `org.eclipse.che.api.core.*Exception`  aka `ServerException ` was made to be used in jax.rs services. In time, it became more popular to use it in internal code. Which is not too bad, but not that it should be. I'm wondering how this method looks like if you push `IOException | JsonSyntaxException ex` to the caller?

Reply:
I see, thank you for letting me know. My latest commit now throws the exception back to the caller and the caller calls it outside of the try block. How does this look?

https://github.com/eclipse-che/che-server/blob/26af7b23bcefebb00ae1ecdd761549a88d780450/multiuser/keycloak/che-multiuser-keycloak-user-remover/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakUserRemover.java#L97-L103